### DISCUSSION TITLE
Defensive Abilities

### DISCUSSION THREAD

**Discussion by mikemangu** [Hekili/hekili#767] [Ideas]
Hey, just wanted to put in this request since I find myself coding it myself on each class I use Hekili on. It would be great if we had a set list of defensive abilities based on situations. Currently, I'm setting up abilities and lock rocks to be used based on health percentages of my character but it would be great if this already came in coded so not to have to redo them every time theres an update. Also, if you can expand on this concept to recommend abilities based on "time to die" or "target count", etc.

**Comment by Hekili** [comment]
https://github.com/Hekili/hekili/wiki/Frequently-Asked-Questions

In the FAQ, you'll see that you should make a copy of the priority so you can preserve your changes with updates.

**Reply by mikemangu** [reply]
Yeah, I get this. I mean when you make an update to the main priority, I have to rewrite the defensive set of actions I already created. It would be helpful if I could export a specific action list to be able to import into an updated main priority.

**Reply by Hekili** [reply]
It would be nice, but I'll probably handle auxiliary action lists differently (i.e., create a separate method for adding utility/defensive/backup filler).

**Reply by romibaer** [reply]
@mikemangu can you pls give an exampel how to add defensives ?
i really would love to add Death Strike to my Unholy Profile 
if HP lower then 65% Death Strike, above Death Coil, if Dark Succor Buff is up => free Death Strike...

can you help ?

**Reply by mikemangu** [reply]
@romibaer yeah, it was fairly simple: I put it as #2 on default action list:

buff.dark_succor.up & health.pct < 50 | health.pct < 25

**Reply by mikemangu** [reply]
Its basically saying, if dark succor is up and health is less than 50% or if health is in general under 25% use Death Strike.
